tests/booleans_test.rs

```diff
@@ -0,0 +1,60 @@
+use uncrustable::syntax::{id, Expr, Map, Type};
+use uncrustable::typecheck::TypeError;
+
+#[test]
+fn test_typeck_bool() {
+    // Create a type environment
+    let env = Map::new();
+
+    // Test boolean literals
+    let true_expr = Expr::Bool(true);
+    let false_expr = Expr::Bool(false);
+
+    // Both should be of type BoolT
+    assert!(matches!(true_expr.typeck_expr(&env), Ok(Type::BoolT)));
+    assert!(matches!(false_expr.typeck_expr(&env), Ok(Type::BoolT)));
+}
+
+#[test]
+fn test_bool_in_context() {
+    // Create a type environment with a boolean variable
+    let mut env = Map::new();
+    env.insert(id("flag"), Type::BoolT);
+
+    // Test variable lookup for boolean variable
+    let flag_expr = Expr::Var(id("flag"));
+    assert!(matches!(flag_expr.typeck_expr(&env), Ok(Type::BoolT)));
+
+    // Test boolean literal in the same context
+    let bool_expr = Expr::Bool(true);
+    assert!(matches!(bool_expr.typeck_expr(&env), Ok(Type::BoolT)));
+}
+
+#[test]
+fn test_undefined_bool_var() {
```

Comment:
just make this a test for `undefined_var`. There is nothing inherent about bools here.

Reply:
I didn't repeat these comments for every single test harness, you should fix others too.